core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -87,10 +87,28 @@ trait InstanceId {
 }
 
 private[chisel3] trait HasId extends InstanceId {
-  private[chisel3] var _parent: Option[BaseModule] = Builder.currentModule
+  // using nullable var for better memory usage
+  private var _parentVar: BaseModule = Builder.currentModule match {
+    case Some(m) => m
+    case None    => null
+  }
+  private[chisel3] def _parent: Option[BaseModule] = Option(_parentVar)
+  private[chisel3] def _parent_=(target: Option[BaseModule]) {
+    _parentVar = target match {
+      case Some(m) => m
+      case None    => null
+    }
+  }
```

Comment:
```suggestion
  private[chisel3] def _parent_=(target: Option[BaseModule]): Unit = {
    _parentVar = target.getOrElse(null)
  }
```
Pithier way to express it, also avoid procedure syntax (always have an `=` when defining a method).